src/index.js

```diff
@@ -5,14 +5,38 @@ import axios from 'axios';
 import { find } from 'lodash';
 import striptags from 'striptags';
 
+function isRunningInExtension() {
+  // * extension API check
+  if (typeof chrome !== 'undefined' && chrome.runtime && chrome.runtime.id) {
+    return true;
+  }
+  if (typeof browser !== 'undefined' && browser.runtime && browser.runtime.id) {
+    return true;
+  }
+
+  // ? Safari, Firefox checks might go here
+
+  return false;
+}
+
+async function fetchData(url) {
+  if (isRunningInExtension()) {
```

Comment:
what about doing the other way round, and check if `fetch` exists, and fall back to axios in other cases (old node, old browser)?